src/main/kotlin/sliv/tool/importer/ProjectParser.kt

```diff
@@ -2,42 +2,77 @@ package sliv.tool.importer
-                images.addAll(folder.listFiles().map {
-                    ImageInfo(it.nameWithoutExtension, it.absolutePath)
-                })
-            } else {
+            if (folder.name != IMAGE_DIRECTORY_NAME) {
+                var splitName = folder.name.split("_")
+                var name = splitName[0]
+                var kind = splitName[1]
+                when (kind) {
+                    "keypoint" -> layers.add(ProjectLayer(LayerKind.KEYPOINT, name))
+                    "plane" -> layers.add(ProjectLayer(LayerKind.PLANE, name))
+                    "line" -> layers.add(ProjectLayer(LayerKind.LINE, name))
+                }
                 folder.listFiles().map {
-                    var key = it.nameWithoutExtension.toLong()
-                    if (files.containsKey(key)) {
-                        files[key]?.add(OutputInfo(folder.nameWithoutExtension, it.absolutePath))
+                    if (landmarks.containsKey(it.nameWithoutExtension.toLong())) {
+                        landmarks[it.nameWithoutExtension.toLong()]?.add(
+                            LandmarkFile(
+                                layers.last(),
+                                Path(it.path),
+                                extractUIDs(it.path)
+                            )
+                        )
                     } else {
-                        files[key] = mutableListOf(OutputInfo(folder.nameWithoutExtension, it.absolutePath))
+                        landmarks.put(
+                            it.nameWithoutExtension.toLong(),
+                            mutableListOf(LandmarkFile(layers.last(), Path(it.path), extractUIDs(it.path)))
+                        )
                     }
                 }
+            } else {
+                folder.listFiles().map {
+                    images.put(it.nameWithoutExtension.toLong(), it.absolutePath)
+                }
             }
         }
+        landmarks.forEach { (name, outputs) ->
+            frames.add(ProjectFrame(name, Path(images[name].toString()), outputs))
+        }
+        return Project(Path(path!!), frames, layers)
+    }
+
+    fun createTreeWithFiles(project: Project, tree: TreeItem<String>): TreeItem<String> {
+        val imgIcon = Image("file:icons/photo.png")
+        val fileIcon = Image("file:icons/description.png")
+
+        var directory = File(project.currentDirectory.toString())
 
         tree.value = directory.name
-        files.map { (name, outputs) ->
-            var imageNode = TreeItem(name.toString())
-            imageNode.children.addAll(outputs.map {
-                TreeItem(it.algo)
+        project.frames.map {
+            var imageNode = TreeItem(it.timestamp.toString(), ImageView(imgIcon))
+            imageNode.children.addAll(it.landmarkFiles.map { landmark ->
+                var layer = landmark.projectLayer
+                var fileName = layer.name + "_" + layer.kind.toString().lowercase()
```

Comment:
maybe show landmarks type by different icons too?

Reply:
I think it can be added later, when implementing error view.